docs/source/developers/cpp/development.rst

```diff
@@ -85,6 +85,8 @@ This project follows `Google's C++ Style Guide
   conveniences.  For private headers the rules are relaxed further and structs
   can be used where convenient for types that do not need access control even
   though they may not be simple data containers.
+* We prefer pointers for non-optional output and input/output parameters (the
+  style guide recommends mutable references).
```

Comment:
Does `non-optional output` refers to an output-only parameter?
If so, then we could say that pointers are preferred for output parameters (optional or required).
Or maybe I am misinterpreting the statement.

Reply:
non-optional is mean to be associative here.  It applies to both non-optional output and non-optional input/output parameters.

Reply:
Thanks for the clarification. English has ambiguous grammar structures.

Reply:
and by associative I meant distributive I think :)  I think the wording could be clarified

Reply:
I was specifically parroting the sentence in the style guide we are overriding (emphasis mine):

> Parameters are either inputs to the function, outputs from the function, or both. Non-optional input parameters should usually be values or const references, **while non-optional output and input/output parameters should usually be references (which cannot be null)**. Generally, use std::optional to represent optional by-value inputs, and use a const pointer when the non-optional form would have used a reference. Use non-const pointers to represent optional outputs and optional input/output parameters.

Without the greater context I agree that we are missing the sentence:

> Use non-const pointers to represent optional outputs and optional input/output parameters.

Would it be clearer if we were to say:

```
We prefer pointers for output and input/output parameters (the
  style guide recommends mutable references in some cases)
```

Reply:
I think just declaring we always prefer pointers is clearer, yeah.

Reply:
Ok, I've changed to that wording.